### PR TITLE
Add doc for alerting comments 

### PR DESCRIPTION
**Pull request by toepkerd**
### Description
Adds documentation for the new experimental feature Alerting Comments. Relevant PR: https://github.com/opensearch-project/alerting/pull/1561

### Issues Resolved
See above PR for issues resolved

### Version
2.15

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Let's hold off on a review until the dev PR gets merged: https://github.com/opensearch-project/alerting/pull/1561

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2108]:
```suggestion
## Update comment
This is an experimental feature and is not recommended for use in a production environment.   
{: .warning}
```

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2206]:
```suggestion
## Delete comment
This is an experimental feature and is not recommended for use in a production environment.   
{: .warning}
```

File: _observing-your-data/alerting/settings.md
Comment (vagimeli) [inline _observing-your-data/alerting/settings.md:63]:
```suggestion
`plugins.alerting.comments_enabled` | `false` | Enables or disables comments for the Alerting plugin.
```

File: _observing-your-data/alerting/settings.md
Comment (vagimeli) [inline _observing-your-data/alerting/settings.md:65]:
```suggestion
`plugins.alerting.comments_history_max_age` | 30d | The oldest document to store in a `.opensearch-alerting-comments-history-<date>` index before creating a new one. If the number of comments in this time period does not exceed `comments_history_max_docs`, one index is created per period (for example, one every 30 days).
```

File: _observing-your-data/alerting/settings.md
Comment (vagimeli) [inline _observing-your-data/alerting/settings.md:66]:
```suggestion
`plugins.alerting.comments_history_rollover_period` | 12h | How often to check if the `.opensearch-alerting-comments-history-write` alias should roll over to a new index and delete old comment history indexes.
```

File: _observing-your-data/alerting/settings.md
Comment (vagimeli) [inline _observing-your-data/alerting/settings.md:67]:
```suggestion
`plugins.alerting.comments_history_retention_period` | 60d | The time to keep comment history indexes before automatic deletion.
```

**Review by vagimeli — APPROVED** [review]
@toepkerd Doc review complete. I made rewrites for clarity and conciseness and alignment with OpenSearch documentation writing style. Those rewrites shouldn't change technical accuracy. Please review and then tag me so that I can move this forward to editorial. Once editorial is complete, I'll address those comments and then merge this PR. Thank you, Melissa

File: _observing-your-data/alerting/comments.md
Comment (vagimeli) [inline _observing-your-data/alerting/comments.md:16]:
```suggestion
When an alert is generated, add comments to share insights into its root cause and facilitate resolution. Comments are enabled by setting `plugins.alerting.comments_enabled` to `true` using the [` _cluster/settings` API]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/settings/).
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:16]:
```suggestion
Query-level monitors run the query and determine whether or not the results should trigger an alert. Query-level monitors can only trigger one alert at a time. For more information about query-level and bucket-level monitors, see [Creating monitors]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/monitors/).
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:84]:
```suggestion
If you use a custom webhook for your destination and need to embed JSON in the message body, be sure to escape the quotation marks:
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:255]:
```suggestion
To specify a time zone, you can do so by including a [cron expression]({{site.url}}{{site.baseurl}}/monitoring-plugins/alerting/cron/) with a time zone name in the `schedule` section of your request. The following example creates a monitor that runs at 12:10 PM Pacific Time on the first day of each month.
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:627]:
```suggestion
To get more specific results in a findings search, you can use any of the optional path parameters defined in the following table.
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:643]:
```suggestion
The following table provides the syntax to use for each trigger option.
```

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:931]: Stacked headings

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:1208]: Stacked headings

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:2387]: Should "comment" be singular (as in the preceding headings)?

File: _observing-your-data/alerting/comments.md
Comment (natebower) [inline _observing-your-data/alerting/comments.md:16]:
```suggestion
When an alert is generated, add comments to share information about its root cause and facilitate resolution. Comments are enabled by setting `plugins.alerting.comments_enabled` to `true` using the [`cluster/settings` API]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/settings/).
```

File: _observing-your-data/alerting/comments.md
Comment (natebower) [inline _observing-your-data/alerting/comments.md:18]:
```suggestion
Comments can be accessed through the alerts table view by selecting the comment icon within an alert's row. From there, comments can be added, edited, or deleted. An Alerting Comments API is also available for programmatic comment management. For more information, see [Alerting API]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/api/).
```

File: _observing-your-data/alerting/comments.md
Comment (natebower) [inline _observing-your-data/alerting/comments.md:22]:
```suggestion
If the Security plugin is installed, then the comment's author is displayed. Otherwise, `Unknown` is displayed.
```

File: _observing-your-data/alerting/comments.md
Comment (natebower) [inline _observing-your-data/alerting/comments.md:26]:
```suggestion
Comment permissions are determined by the backend roles associated with the alert. These backend roles are inherited from the monitor that generated the alert. For more information about how to limit access based on backend roles, see [Limit access by backend role]({{site.url}}{{site.baseurl}}/observing-your-data/alerting/security/#advanced-limit-access-by-backend-role).
```

File: _observing-your-data/alerting/settings.md
Comment (natebower) [inline _observing-your-data/alerting/settings.md:65]:
```suggestion
`plugins.alerting.comments_history_max_age` | 30d | The oldest document to store in an `.opensearch-alerting-comments-history-<date>` index before creating a new one. If the number of comments in the specified time period does not exceed `comments_history_max_docs`, then 1 index is created per period (for example, 1 every 30 days).
```

File: _observing-your-data/alerting/settings.md
Comment (natebower) [inline _observing-your-data/alerting/settings.md:66]:
```suggestion
`plugins.alerting.comments_history_rollover_period` | 12h | How often to determine whether the `.opensearch-alerting-comments-history-write` alias should roll over to a new index and delete old comment history indexes.
```

File: _observing-your-data/alerting/settings.md
Comment (natebower) [inline _observing-your-data/alerting/settings.md:67]:
```suggestion
`plugins.alerting.comments_history_retention_period` | 60d | The amount of time to keep comment history indexes before automatic deletion.
```

Comment (natebower) [inline _observing-your-data/alerting/settings.md:67]: "keep" => "store"?

File: _observing-your-data/alerting/settings.md
Comment (natebower) [inline _observing-your-data/alerting/settings.md:70]:
```suggestion
`plugins.alerting.max_comments_per_notification` | 3 | The maximum number of comments per alert to include in the `ctx` Mustache template variable for alert notifications.
```

**Review by natebower — COMMENTED** [review]
@vagimeli @toepkerd Please see my comments and changes and let me know if you have any questions. Thanks!

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:746]:
```suggestion
When updating a monitor, you can optionally include `seq_no` and `primary_term` as URL parameters. If these numbers do not match the existing monitor or the monitor does not exist, the Alerting plugin throws an error. OpenSearch increments the version number and the sequence number automatically (see the example response).
```

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [reply _observing-your-data/alerting/api.md:1211]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1249]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1526]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1774]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1815]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1914]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1959]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:1998]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2075]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2173]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2218]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2257]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2335]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2377]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2420]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (vagimeli) [inline _observing-your-data/alerting/api.md:2502]: @natebower Added intro sentence.

File: _observing-your-data/alerting/api.md
Comment (natebower) [inline _observing-your-data/alerting/api.md:2335]:
```suggestion
Add comments to a specific alert, providing additional context or notes related to that alert, using the following request.
```

**Review by natebower — APPROVED** [review]
@vagimeli LGTM with one minor change. Thanks!

**Pull request merged**
